packages/api-gateway/src/services/tenantService.ts

```diff
@@ -36,3 +41,119 @@ export async function getOrganization(
+        headers,
+        tenantId
+      );
+    },
+    getOrganizationEservices: async (
+      { logger, headers }: WithLogger<ApiGatewayAppContext>,
+      params: {
+        origin: tenantApi.ExternalId["origin"];
+        externalId: tenantApi.ExternalId["value"];
+        attributeOrigin: string;
+        attributeCode: string;
+      }
+    ): Promise<apiGatewayApi.EServices> => {
+      const { origin, externalId, attributeOrigin, attributeCode } = params;
+      logger.info(
+        `Retrieving Organization EServices for origin ${origin} externalId ${externalId} attributeOrigin ${attributeOrigin} attributeCode ${attributeCode}`
+      );
+
+      const tenant = await tenantProcessClient.tenant.getTenantByExternalId({
+        headers,
+        params: {
+          origin,
+          code: externalId,
+        },
+      });
+
+      const attribute =
+        await attributeProcessClient.getAttributeByOriginAndCode({
+          headers,
+          params: {
+            origin: attributeOrigin,
+            code: attributeCode,
+          },
+        });
+
+      const allEservices = await getAllEservices(
+        catalogProcessClient,
+        headers,
+        tenant.id,
+        attribute.id
+      );
+      const allowedEservices = allEservices.filter(
+        (eservice) => eservice.descriptors.length > 0
+      );
+
+      const eservices = await Promise.all(
+        allowedEservices.map((eservice) =>
+          enhanceEservice(
+            tenantProcessClient,
+            attributeProcessClient,
+            headers,
+            eservice
+          )
+        )
+      );
+      return { eservices };
+    },
+    revokeTenantAttribute: async (
+      { logger, headers }: WithLogger<ApiGatewayAppContext>,
+      params: {
+        origin: tenantApi.ExternalId["origin"];
+        externalId: tenantApi.ExternalId["value"];
+        attributeCode: tenantApi.M2MAttributeSeed["code"];
+      }
+    ): Promise<void> => {
+      const { origin, externalId, attributeCode } = params;
+      logger.info(
+        `Revoking attribute ${attributeCode} of tenant (${origin},${externalId})`
+      );
+
+      await tenantProcessClient.m2m.m2mRevokeAttribute(undefined, {
+        headers,
+        params: {
+          origin,
+          externalId,
+          code: attributeCode,
+        },
+      });
+    },
+    upsertTenant: async (
+      { logger, headers }: WithLogger<ApiGatewayAppContext>,
+      params: {
+        origin: tenantApi.ExternalId["origin"];
+        externalId: tenantApi.ExternalId["value"];
+        attributeCode: tenantApi.M2MAttributeSeed["code"];
+      }
+    ): Promise<void> => {
+      const { origin, externalId, attributeCode } = params;
+      logger.info(
+        `Upserting tenant with externalId (${origin},${externalId}) with attribute ${attributeCode}`
+      );
+
+      const tenantSeed = toM2MTenantSeed(origin, externalId, attributeCode);
+      await tenantProcessClient.m2m.m2mUpsertTenant(tenantSeed, {
```

Comment:
It was not implemented yet in the tenant process - still, it's in the open API, so the client is generated correctly and I was able to implement it on the API GW side, but not to test it yet.